ocs_ci/deployment/hosted_cluster.py

```diff
@@ -660,6 +745,25 @@ def create_host_inventory():
     logger.info("Created InfraEnv.")
 
 
+def get_onboarding_token_from_secret(secret_name):
+    """
+    Get onboarding token from the secret
+
+    Args:
+        secret_name (str): Name of the secret
+
+    Returns:
+        str: Onboarding token
+    """
+    ocp_obj = OCP(
+        kind="secret",
+        namespace=config.ENV_DATA["cluster_namespace"],
+        resource_name=secret_name,
+    )
+    secret_obj = ocp_obj.get(retry=6, wait=10, silent=True)
+    return secret_obj.get("data").get("onboarding-token")
+
+
```

Comment:
So we will be able to only fetch the onboarding token from the secret ? Generate onboarding token option will not be available?

Reply:
Based on how I understand Leela's words this is by design starting from 4.17. We did onboarding incorrectly, generating secret by our own. But I did not check on older versions